utils.js

```diff
@@ -0,0 +1,35 @@
+class RecentlyUsedList {
+
+    constructor() {
+        this.items = [];
+    }
+
+    add(newItem) {
+        if (this.items.indexOf(newItem)) {
+            var position = this.items.indexOf(newItem);
+            var existingItem = this.items[position];
+
+            this.items.splice(position, 1);
+            this.items.splice(0, 0, existingItem);
+        } else {
+            this.items.unshift(newItem);
+        }
+    };
+
+    lenght() {
```

Comment:
spelling, also `length` is a basic property of an object, it's not necessary to create a function like this.